### PR TITLE
BUG: shows correct package name when import_optional_dependency is ca…

### LINKED ISSUE #35948: DOC: "engine=odf" option should ask you to install pyodf instead of odf in pandas.read_excel
#### Location of the documentation

"pandas.read_excel"

#### Documentation problem

When `engine="odf"` is used and `pyodf` is not installed it gives an error that you should install optional dependency odf as `pip or conda install odf`. This pip/conda command fails because there's no module called `odf`. This could also be a security vulnerability if someone puts a malicious "odf" module on pypi.

#### Suggested fix for documentation

The error should ask you to install odf using `pip or conda install pyodf`

### PR DESCRIPTION
**Pull request by hs2361**
…lled. fixes #35948

- [x] closes #35948
- [x] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/compat/_optional.py [hunk 1]
```diff
@@ -33,6 +33,32 @@
     "numba": "0.46.0",
 }
 
+INSTALL_MAPPING = {
```
Comment (jreback) [inline pandas/compat/_optional.py:36]: only add the names of packages that are *different* (as we already have all of the names); add a comment to describe this.

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a note in whatsnew 1.1.2 Other bug fix section is fine

File: pandas/compat/_optional.py (hunk 1 above)
Comment (hs2361) [reply pandas/compat/_optional.py:39]: I have made the changes as suggested.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green

**Comment by hs2361** [comment]
@jreback All checks are green!

**Comment by hs2361** [comment]
@jreback Could you please help me to figure out why the Travis CI build is failing?

**Comment by simonjayhawkins** [comment]
I've restarted travis to see if failure persists.

**Comment by simonjayhawkins** [comment]
will be merging and backporting if green

**Comment by hs2361** [comment]
@simonjayhawkins The Travis CI build passed this time. Thanks for your help.

**Comment by simonjayhawkins** [comment]
Thanks @hs2361

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by simonjayhawkins** [comment]
maybe due to release notes out of sync https://github.com/pandas-dev/pandas/issues/35831#issuecomment-688174481

i'll resolve that and then see if the backport can be triggered a second time.

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**